### ISSUE TITLE
Subsystems and info files

### ISSUE THREAD

**Issue by zach-morris** [libretro/fbalpha#130]
Trying the new builds with updated neogeocd support.  Not sure if it's too early to be writing issues or not.

I'm unable to start a game with the error given:
[ERROR] failed_to_start_audio_driver

I've tried a redump game (single bin/cue) and trurip (ccd/sub/img) and both fail for the same reason.

Redump single bin / cue contents:
```
FILE "Crossed Swords (Japan).bin" BINARY
  TRACK 01 MODE1/2352
    INDEX 01 00:00:00
  TRACK 02 AUDIO
    INDEX 00 00:42:12
    INDEX 01 00:45:12
  TRACK 03 AUDIO
    INDEX 01 01:29:12
  TRACK 04 AUDIO
    INDEX 01 02:32:58
  TRACK 05 AUDIO
    INDEX 01 03:17:58
  TRACK 06 AUDIO
    INDEX 01 04:21:74
  TRACK 07 AUDIO
    INDEX 01 06:12:74
  TRACK 08 AUDIO
    INDEX 01 06:42:74
  TRACK 09 AUDIO
    INDEX 01 08:08:14
  TRACK 10 AUDIO
    INDEX 01 08:51:15
  TRACK 11 AUDIO
    INDEX 01 10:53:46
  TRACK 12 AUDIO
    INDEX 01 12:02:46
  TRACK 13 AUDIO
    INDEX 01 12:39:46
  TRACK 14 AUDIO
    INDEX 01 13:19:46
  TRACK 15 AUDIO
    INDEX 01 13:42:46
  TRACK 16 AUDIO
    INDEX 01 14:00:46
  TRACK 17 AUDIO
    INDEX 01 16:51:01
  TRACK 18 AUDIO
    INDEX 01 20:34:16
  TRACK 19 AUDIO
    INDEX 01 23:17:31
  TRACK 20 AUDIO
    INDEX 01 26:33:01
  TRACK 21 AUDIO
    INDEX 01 30:12:46
  TRACK 22 AUDIO
    INDEX 01 33:54:46
  TRACK 23 AUDIO
    INDEX 01 37:52:16
  TRACK 24 AUDIO
    INDEX 01 40:58:62
  TRACK 25 AUDIO
    INDEX 01 44:35:32
  TRACK 26 AUDIO
    INDEX 01 48:12:02
  TRACK 27 AUDIO
    INDEX 01 48:54:02
```

Trurip cue contents:
```
REM Ripping Tool: trurip

FILE "Neo DriftOut (JP).img" BINARY
  TRACK 01 MODE1/2352
    INDEX 01 00:00:00
  TRACK 02 AUDIO
    INDEX 00 01:18:62
    INDEX 01 01:20:62
  TRACK 03 AUDIO
    INDEX 00 04:32:65
    INDEX 01 04:34:65
  TRACK 04 AUDIO
    INDEX 00 07:46:69
    INDEX 01 07:48:69
  TRACK 05 AUDIO
    INDEX 00 11:00:72
    INDEX 01 11:02:72
  TRACK 06 AUDIO
    INDEX 00 14:06:73
    INDEX 01 14:08:73
  TRACK 07 AUDIO
    INDEX 00 17:29:28
    INDEX 01 17:31:28
  TRACK 08 AUDIO
    INDEX 00 20:43:33
    INDEX 01 20:45:33
  TRACK 09 AUDIO
    INDEX 00 23:57:36
    INDEX 01 23:59:36

REM DISC HASHES
REM CRC32 : 8572e8a4
REM MD5   : 1ac8b0c09825e787e4d428463fd2267d
REM SHA1  : 5903156474180018b9852fee97613763fdbbfc6d

REM Gap Append Method: Prev [crc32]
REM Trk 01: 39dc8c96
REM Trk 02: 438b4190
REM Trk 03: 8d821809
REM Trk 04: 963626c4
REM Trk 05: 3f2683dc
REM Trk 06: 08b15936
REM Trk 07: c20221c4
REM Trk 08: d788dbbd
REM Trk 09: f9c9d886

REM Gap Append Method: Prev [md5]
REM Trk 01: 42495bf93c0cb3e483b3733dfb6f97eb
REM Trk 02: acbfe345543164541d81b0631753dca8
REM Trk 03: 4c4be064eac60c210efc5e4356b18561
REM Trk 04: d96abb5fa00456d4a93301b0f715c7df
REM Trk 05: f5e3b09928e2fb37bffedfd9b7cd38a1
REM Trk 06: 2cd3fd432ad8204942cc73b4103d5cce
REM Trk 07: b85fa020bb2ebe8874de82cecdb96fc1
REM Trk 08: 770a17d6f750223ecdb5b9a54305eb52
REM Trk 09: 7f2799749aab184954c7a2fc2fe6c9db

REM Gap Append Method: Prev [sha1]
REM Trk 01: d3cdf8c990e10cc6019ba3763d777554089e7f53
REM Trk 02: 992f591379f64e1fb81d4a2e9075742a6774aad7
REM Trk 03: 5a93a5341d9ad7ae86d6d00325188ee1a108271d
REM Trk 04: 1a722f994edf5330cc4e87b563a0f3e1dd534afe
REM Trk 05: b4906199073f1bdd9045d3725b9894162b9a424a
REM Trk 06: 766a21cfc8674e9b576cd5d975d052666c991fad
REM Trk 07: 1d96a98c056f111c04ee9023b3b608f2a299cbb8
REM Trk 08: 154c1dccb7fafeef613d4aa1b92f2b539cedeacf
REM Trk 09: 567b81628be10b4eb3e9cb3987196f73d4ded202

REM Gap Append Method: None [crc32]
REM Trk 01: d2dc97cd
REM Trk 02: 467553f3
REM Trk 03: c2d43c60
REM Trk 04: 94605059
REM Trk 05: 3e26b42f
REM Trk 06: 7c09cbf6
REM Trk 07: 2b36c44b
REM Trk 08: 47446e15
REM Trk 09: f9c9d886
REM Gap 02: 7abe247f
REM Gap 03: 7abe247f
REM Gap 04: 7abe247f
REM Gap 05: 7abe247f
REM Gap 06: 7abe247f
REM Gap 07: 7abe247f
REM Gap 08: 7abe247f
REM Gap 09: 7abe247f

REM Gap Append Method: None [md5]
REM Trk 01: ec16dacaf4c14cce80a3c08f9ff32880
REM Trk 02: 30e7f2856d1ff6ac5224069e139a6a89
REM Trk 03: 29446f15e3322fd44f120d687d64cc88
REM Trk 04: 4a1b785191a839d2b8ede4fac11b85e7
REM Trk 05: 359d4d4f1c5815a8e5fafa7863b15991
REM Trk 06: 37053fc988802e0a7807080d370dd83e
REM Trk 07: 66c7126309a8f471c6595d6012b1337f
REM Trk 08: eec8297d06823bf4b61d19bbfb2fa841
REM Trk 09: 7f2799749aab184954c7a2fc2fe6c9db
REM Gap 02: 9df204fbc98bb8965081cb30a89715fc
REM Gap 03: 9df204fbc98bb8965081cb30a89715fc
REM Gap 04: 9df204fbc98bb8965081cb30a89715fc
REM Gap 05: 9df204fbc98bb8965081cb30a89715fc
REM Gap 06: 9df204fbc98bb8965081cb30a89715fc
REM Gap 07: 9df204fbc98bb8965081cb30a89715fc
REM Gap 08: 9df204fbc98bb8965081cb30a89715fc
REM Gap 09: 9df204fbc98bb8965081cb30a89715fc

REM Gap Append Method: None [sha1]
REM Trk 01: 9403e89f77a0c52511e11be01718660f3081a47f
REM Trk 02: a50f4279a63d93d49f38284d73325b6401e76f6b
REM Trk 03: cf51c780e7529ae31bfe9f6f7c5e8b16b15dd49b
REM Trk 04: aca6c5f2b7366b06d74e6b1406146dc617a91459
REM Trk 05: e195cde3249a2d65eafc511830275f068199f1f5
REM Trk 06: 391fec4203237379d85539164f9a4b01b9764ccf
REM Trk 07: 3ddb9f9893f35846f8bd9b0c949d6ec289098a26
REM Trk 08: 434108789a28836c5ec18f90002bcfb12ba685ed
REM Trk 09: 567b81628be10b4eb3e9cb3987196f73d4ded202
REM Gap 02: cbe29e0bd725ef1194f8c47619dd42078e841de2
REM Gap 03: cbe29e0bd725ef1194f8c47619dd42078e841de2
REM Gap 04: cbe29e0bd725ef1194f8c47619dd42078e841de2
REM Gap 05: cbe29e0bd725ef1194f8c47619dd42078e841de2
REM Gap 06: cbe29e0bd725ef1194f8c47619dd42078e841de2
REM Gap 07: cbe29e0bd725ef1194f8c47619dd42078e841de2
REM Gap 08: cbe29e0bd725ef1194f8c47619dd42078e841de2
REM Gap 09: cbe29e0bd725ef1194f8c47619dd42078e841de2

REM Gap Append Method: Next [crc32]
REM Trk 01: d2dc97cd
REM Trk 02: a71ceb00
REM Trk 03: b915d5b5
REM Trk 04: 7509e8aa
REM Trk 05: e9db65c3
REM Trk 06: ea40593d
REM Trk 07: 80634ada
REM Trk 08: a62dd6e6
REM Trk 09: 18a06075

REM Gap Append Method: Next [md5]
REM Trk 01: ec16dacaf4c14cce80a3c08f9ff32880
REM Trk 02: aea70d2999ecb424296a8dd1609cc6cc
REM Trk 03: 768e8df79db95dcc6e58ea8120392e11
REM Trk 04: 3184888fa8b829a2c9a6cf5ff2086930
REM Trk 05: 01a546f0d592ead32f1ebaf139b599e2
REM Trk 06: 9bfb78150642020c6b8b7c1de9c96079
REM Trk 07: 6048bbb125e59c696968f92d3eb10cae
REM Trk 08: 781d9464c4946d15abc016dba9cdb4c7
REM Trk 09: 90f2a430910eddc5faebe16b911beeab

REM Gap Append Method: Next [sha1]
REM Trk 01: 9403e89f77a0c52511e11be01718660f3081a47f
REM Trk 02: b64ee63868f00ba35c31b1650791cc24cfd00311
REM Trk 03: b220a3b6ee2062522dd6025992590d63b305526f
REM Trk 04: 1fd6186b7341bd7725bee48a4d5f2d58524351b0
REM Trk 05: 54f75af93e34c0caa5fd0e49303cb098158628f6
REM Trk 06: 4fdd82648cb53eb173e37aa4af13cd6551cc5146
REM Trk 07: f439803b9f7a06ef71fff22be6bba4d3e6a86d29
REM Trk 08: c80c4a4092f35954d9e90542dab1db4cd43585c5
REM Trk 09: d0d6939c07782ecde63f14bd0448ae86eeb67edd
```

**Comment by barbudreadmon** [comment]
Could you please give me your complete log ? I'm not experiencing this issue, and you shouldn't either, your isos seem fine.

**Comment by zach-morris** [comment]
Thanks, here's the complete log for the redump single bin/cue:
```
xxxs-MacBook-Pro:~ xxx$ "/Applications/RetroArch.app/Contents/MacOS/RetroArch"  -L "/Users/xxx/Documents/RetroArch/cores/fbalpha_libretro.dylib" "/Volumes/Untitled/NGCD/Crossed Swords (Japan)/Crossed Swords (Japan).cue"
[INFO] [recording] twitch streaming key empty[INFO] Redirecting save file to "/Users/xxx/Documents/RetroArch/saves/Crossed Swords (Japan).srm".
[INFO] Redirecting savestate to "/Users/xxx/Documents/RetroArch/states/Crossed Swords (Japan).state".
[INFO] === Build =======================================
Capabilities: MMX MMXEXT SSE1 SSE2 SSE3 SSSE3 SSE4 SSE4.2 AVX AVX2 AES 
Built: Oct 22 2018
[INFO] Version: 1.7.5
[INFO] =================================================
[INFO] Loading dynamic libretro core from: "/Users/xxx/Documents/RetroArch/cores/fbalpha_libretro.dylib"
[INFO] [overrides] no core-specific overrides found at /Users/xxx/Library/Application Support/RetroArch/config/FB Alpha/FB Alpha.cfg.
[INFO] [overrides] no content-dir-specific overrides found at /Users/xxx/Library/Application Support/RetroArch/config/FB Alpha/Crossed Swords (Japan).cfg.
[INFO] [overrides] no game-specific overrides found at /Users/xxx/Library/Application Support/RetroArch/config/FB Alpha/Crossed Swords (Japan).cfg.
[INFO] Shaders: preset directory: /Applications/RetroArch.app/Contents/Resources/shaders/presets
[INFO] Shaders: no game-specific preset found at /Applications/RetroArch.app/Contents/Resources/shaders/presets/FB Alpha/Crossed Swords (Japan).cgp.
[INFO] Shaders: no game-specific preset found at /Applications/RetroArch.app/Contents/Resources/shaders/presets/FB Alpha/Crossed Swords (Japan).glslp.
[INFO] Shaders: no content-dir-specific preset found at /Applications/RetroArch.app/Contents/Resources/shaders/presets/FB Alpha/Crossed Swords (Japan).cgp.
[INFO] Shaders: no content-dir-specific preset found at /Applications/RetroArch.app/Contents/Resources/shaders/presets/FB Alpha/Crossed Swords (Japan).glslp.
[INFO] Shaders: no core-specific preset found at /Applications/RetroArch.app/Contents/Resources/shaders/presets/FB Alpha/FB Alpha.cgp.
[INFO] Shaders: no core-specific preset found at /Applications/RetroArch.app/Contents/Resources/shaders/presets/FB Alpha/FB Alpha.glslp.
[INFO] Environ SET_SUBSYSTEM_INFO.
[INFO] Special game type: CBS ColecoVision
[INFO]   Ident: cv
[INFO]   ID: 1
[INFO]   Content:
[INFO]     Rom (required)
[INFO] Special game type: MSX 1
[INFO]   Ident: msx
[INFO]   ID: 4
[INFO]   Content:
[INFO]     Rom (required)
[INFO] Special game type: Nec PC-Engine
[INFO]   Ident: pce
[INFO]   ID: 5
[INFO]   Content:
[INFO]     Rom (required)
[INFO] Special game type: Nec SuperGrafX
[INFO]   Ident: sgx
[INFO]   ID: 7
[INFO]   Content:
[INFO]     Rom (required)
[INFO] Special game type: Nec TurboGrafx-16
[INFO]   Ident: tg16
[INFO]   ID: 9
[INFO]   Content:
[INFO]     Rom (required)
[INFO] Special game type: Sega GameGear
[INFO]   Ident: gg
[INFO]   ID: 2
[INFO]   Content:
[INFO]     Rom (required)
[INFO] Special game type: Sega Master System
[INFO]   Ident: sms
[INFO]   ID: 8
[INFO]   Content:
[INFO]     Rom (required)
[INFO] Special game type: Sega Megadrive
[INFO]   Ident: md
[INFO]   ID: 3
[INFO]   Content:
[INFO]     Rom (required)
[INFO] Special game type: Sega SG-1000
[INFO]   Ident: sg1k
[INFO]   ID: 6
[INFO]   Content:
[INFO]     Rom (required)
[INFO] Special game type: ZX Spectrum
[INFO]   Ident: spec
[INFO]   ID: 10
[INFO]   Content:
[INFO]     Rom (required)
[INFO] Special game type: Neogeo CD
[INFO]   Ident: neocd
[INFO]   ID: 11
[INFO]   Content:
[INFO]     Iso (required)
[INFO] Remaps: remap directory: /Users/xxx/Library/Application Support/RetroArch/config/remaps
[INFO] Remaps: no game-specific remap found at /Users/xxx/Library/Application Support/RetroArch/config/remaps/FB Alpha/Crossed Swords (Japan).rmp.
[INFO] Remaps: no content-dir-specific remap found at /Users/xxx/Library/Application Support/RetroArch/config/remaps/FB Alpha/Crossed Swords (Japan).rmp.
[INFO] Remaps: no core-specific remap found at /Users/xxx/Library/Application Support/RetroArch/config/remaps/FB Alpha/FB Alpha.rmp.
[INFO] Redirecting save file to "/Users/xxx/Documents/RetroArch/saves/Crossed Swords (Japan).srm".
[INFO] Redirecting savestate to "/Users/xxx/Documents/RetroArch/states/Crossed Swords (Japan).state".
[INFO] Environ GET_LOG_INTERFACE.
[INFO] Content loading skipped. Implementation will load it on its own.
[INFO] CRC32: 0x3f3f9a4d .
[libretro INFO] Setting save dir to /Users/xxx/Documents/RetroArch/saves
[INFO] Environ SYSTEM_DIRECTORY: "/Users/xxx/Documents/RetroArch/system".
[libretro INFO] Setting system dir to /Users/xxx/Documents/RetroArch/system
[ERROR] Failed to load content.
[INFO] Unloading game..
[INFO] Unloading core..
[INFO] Unloading core symbols..
[INFO] [Video]: Does not have enough samples for monitor refresh rate estimation. Requires to run for at least 4096 frames.
[ERROR] failed_to_start_audio_driver
```

**Comment by zach-morris** [comment]
I'm an idiot.  I forgot the --subsystem neocd argument.  Sorry to bother

**Comment by barbudreadmon** [comment]
> The 'cue' file extension doesn't seem to be listed as an option in the info file for fbalpha, so it doesn't show up as something that can be loaded from the GUI.

I might be wrong but i don't think this stuff works with subsystems, all it would do is ask if you want to use fbalpha, however it would fail loading the game because fbalpha needs the subsystem setting to run those games. It would end up being even more confusing for the user.
Maybe the info stuff could have some improvement to handle subsystems ? Let's tag @fr500 about this, he might know something.

**Comment by zach-morris** [comment]
So I tried this:
Updated the info file from
```
supported_extensions = "iso|zip|7z"
```
to
```
supported_extensions = "cue|zip|7z"
```

From the gui, selected load content, pointed to the cue, started with FBAlpha.  Failed to load, but then I could get the subsystem listings from the gui.  Selected Load Neo Geo CD, and pointed to the cue file again, and it loaded.  This is pretty much the same functionality that was there prior, so maybe it's just a matter of updating the info file (not sure if iso format is even applicable anymore)
